libunwind/src/UnwindRegistersRestore.S

```diff
@@ -20,7 +20,7 @@
   .text
 #endif
 
-#if !defined(__USING_SJLJ_EXCEPTIONS__)
+#if !defined(__USING_SJLJ_EXCEPTIONS__) && !defined(__wasm__)
```

Comment:
You are defining `__USING_WASM_EXCEPTIONS__`. Why you are not using it here and everywhere else?

Reply:
`__wasm__` is defined whenever we compile the code to wasm, and `__USING_WASM_EXCEPTIONS__` is defined when we compile the code to wasm and use Wasm exceptions (`-fwasm-exeptions`). And this code should be excluded whenever it is compiled to wasm, even if `-fwasm-exceptions` is not used.

Reply:
Ok. Then does the clang change really belong here? `LIBUNWIND_USES_WASM_EXCEPTIONS` also seems to be unused.

Reply:
Removed the clang change and `LIBUNWIND_USES_WASM_EXCEPTIONS`.

Reply:
Can I land this? Thank you for your review!